Assets/Scripts/UI/Canvas/PhotoBookSystem/PhotoBookWindow.cs

```diff
@@ -0,0 +1,268 @@
+﻿using UnityEditor;
+using UnityEngine;
+using System.IO;
+using System.Text.RegularExpressions;
+
+using Framework.GeoLocation;
+using Framework.PhoneCamera;
+using Framework.ScriptableObjects;
+using UI.Canvas.PhotoTaking;
+
+namespace UI.Canvas.PhotoBookSystem
+{
+    public sealed class PhotoBookWindow : EditorWindow
+    {
+        private const int MARGIN = 20;
+
+        private const string SPACE = " ";
+        private const string INTERACTABLE_PATH = "Assets/Prefabs/Funcinal/Entities/Photo-Interactable.prefab";
+        private const string PAGE_PAGE = "Assets/Prefabs/Funcinal/UI/Canvas/PhotoBook/Page.prefab";
+        private const string DIRECTORY_PATH = "Assets/ScriptableObjects/Photos/";
+        private const string PHOTO_BOOK_NAME = "Photobook";
+        private const string PHOTO_MAKER_NAME = "Photo maker";
+        private const string CAMERA_PANEL_NAME = "CameraPanel";
+        private const string NAME_FIELD = "Name for new page:";
+        private const string INFO_FIELD = "Info for new page:";
+        private const string CORDS_FIELD = "Vector2 coordinates field:";
+        private const string TEXTURE_FIELD = "Render texture for interactable ->";
+        private const string BUTTON_TEXT = "Make new page!";
+        private const string SCRIPTABLE_OBJECT_SUFFIX = ".asset";
+        private const string INVALID_CHARACTERS = "^[a-zA-Z0-9 ]+$";
+        private const string PREFAB_PATH_NOT_FOUND_ERROR = "Prefab not found at path: ";
+        private const string NAME_NEEDED_ERROR = "There is a name needed!";
+        private const string INVALID_NAME_ERROR = "Invalid input for the name field.";
+        private const string PHOTO_BOOK_REFERENCE_WARNING = "You need to reference the page in the photobook pages list!";
+        private const string TEXTURE_WARNING = "Do not click on this, only drag and drop a texture! ^^";
+        private const string PHOTO_INTERACTABLE_REFERENCE_WARNING = "You need to reference the interactable in the page!";
+        
+        private static readonly Vector2 WINDOW_SIZE = new (400f, 375f);
+        
+        private bool _needsRepaint;
+        private string _newObjectName = "Default Text";
+        private string _info = "Info";
+        private Vector2 _cords = Vector2.zero;
+        private Texture2D _interactableTexture;
+        private Page _currentPage;
+        private PhotoData _currentData;
+        private GUIStyle _greenButtonStyle;
+
+        [MenuItem("Codename-BDLMTW/PhotoBook Window")]
+        public static void ShowWindow()
```

Comment:
Summary nodig?

Reply:
Niet nodig, je kan dit niet in code aan roepen. Het staat bovenin Unity met een dropdown.